### DISCUSSION TITLE
Playlist on quit/reopen: it remembers the list but jumps to the next item

### DISCUSSION THREAD

**Discussion by dymky** [smplayer-dev/smplayer#789] [Q&A] [answered]
If I quit and open again the SMPlayer with an active playlist and "Save a copy of the playlist on exit" option enabled, then starting to play jumps to the next entry in the playlist (even though the previous one has not been finished yet).

If I disable "Save a copy of the playlist on exit", then the list isn't remembered at all.

This is SMPlayer 23.6.0, 10157, Ubuntu.

What am I doing wrong? Thank you!

**Comment by Raza-0007** [comment, marked as answer]
I use playlists all the time and have never experienced this issue. I currently have the dynamic playlist option enabled, and everything is working as expected. Here are my settings:


![Playlist 1](https://github.com/smplayer-dev/smplayer/assets/131807929/8e623388-7d22-419d-b84d-06367430f0a3)

![Playlist 2](https://github.com/smplayer-dev/smplayer/assets/131807929/2e90e21a-1d9d-48c4-8387-35c663fe1e92)

Try and copy my settings and see if the issue persists. If it does then next step will be to reset smplayer to default settings. On Windows this can be achieved by removing the smplayer.ini from your smplayer folder, I am assuming it is the same for Ubuntu. Please backup everything as you will lose all your customized settings after this.

~Raza

**Reply by dymky** [reply]
Thank you for the answer!

The problem was in either ~/.config/smplayer/player_info.ini or ~/.config/smplayer/playlist.ini: nothing else had helped (reinstalling, changing versions, removing "smplayer.ini", etc) until I deleted those two files and restarted the player.

**Reply by Raza-0007** [reply]
I am glad it worked. It was most likely playlist.ini, as smplayer stores the current playlist and its playlist settings there, and that also had to be removed to restore it to default.

**Reply by dymky** [reply]
Right. By the way, after it got fixed, I put back my previous "smplayer.ini" (and things still work as they should).

**Reply by Raza-0007** [reply]
That is even better. Then it was definitely playlist.ini that was causing the issue. All the settings in it are the settings under the playlist tab in smplayer preferences, but sometime these .ini files get corrupted, and the only way to recover is to delete them and start fresh.

**Reply by dymky** [reply]
I hope I am not violating the rules by asking a related "further question".

Is it possible to emulate "bookmarks behaviour" for playlists?
Say, I want to have two different playlists and be able to easily switch between them, automatically coming back to the recently played item in each list (e.g., imagine switching between several audio-books)?

If not supported natively, can this be implemented via some "file-substituting" script?

Thanks!

**Reply by Raza-0007** [reply]
>Is it possible to emulate "bookmarks behaviour" for playlists?

I do not think it is possible within smplayer. Anything like this will have to be baked-in within smplayer code. 

You maybe able to find a third party script that implements this feature in mpv, but you will have to use mpv directly without smplayer. If interested [here is great list of mpv scripts](https://github.com/stax76/awesome-mpv#user-script).

**Reply by dymky** [reply]
Thank you, I'll have a look!
